Wikipedia/Code/WMFRandomArticleFetcher.m

```diff
@@ -66,33 +66,34 @@ - (MWKSearchResult*)getBestRandomResultFromResults:(NSArray*)results {
     //Sort so random results with good extracts and images come first and disambiguation pages come last.
     NSSortDescriptor* extractSorter  = [[NSSortDescriptor alloc] initWithKey:@"extract.length" ascending:NO];
     NSSortDescriptor* descripSorter  = [[NSSortDescriptor alloc] initWithKey:@"wikidataDescription" ascending:NO];
-    NSSortDescriptor* thumbSorter    = [[NSSortDescriptor alloc] initWithKey:@"thumbnailURL" ascending:NO];
+    NSSortDescriptor* thumbSorter    = [[NSSortDescriptor alloc] initWithKey:@"thumbnailURL.absoluteString" ascending:NO];
     NSSortDescriptor* disambigSorter = [[NSSortDescriptor alloc] initWithKey:@"isDisambiguation" ascending:YES];
     NSSortDescriptor* listSorter     = [[NSSortDescriptor alloc] initWithKey:@"isList" ascending:YES];
-    results = [results sortedArrayUsingDescriptors:@[disambigSorter, listSorter, extractSorter, thumbSorter, descripSorter]];
+    results = [results sortedArrayUsingDescriptors:@[disambigSorter, listSorter, thumbSorter, descripSorter, extractSorter]];
```

Comment:
Extract length descriptor was moved to the end so it can act as a tie breaker for random items which have both an image and a wikidata description. 

Having it first was meaningless as the other sort criteria would only really take effect if 2 random results had extracts of the exact same length, which is unlikely.